core/src/test/scala/tech/beshu/ror/accesscontrol/blocks/definitions/ldap/implementations/ReleasablePoolTest.scala

```diff
@@ -0,0 +1,135 @@
+import cats.implicits._
+import monix.eval.Task
+import monix.execution.schedulers.TestScheduler
+
+import concurrent.duration._
+import language.postfixOps
+import language.higherKinds
+import org.scalatest.Matchers._
+
+import scala.concurrent.Await
+private final class ReleasablePoolTaskTest extends ReleasablePoolTest[Task] {
+  implicit val scheduler: TestScheduler= TestScheduler()
+  override protected def acquire(counter: Counter): Task[counter.ReleasableResource] = Task.evalAsync(counter.create)
+  override protected def release(counter: Counter)(resource: counter.ReleasableResource): Task[Unit] = Task.evalAsync(resource.close)
+  override protected def await[A](a: Task[A]): A = {
+    val t = a.executeAsync.runToFuture
+    scheduler.tick()
+    Await.result(t, 2 second)
+  }
+}
+private final class ReleasablePoolIdTest extends ReleasablePoolTest[Id] {
+  override protected def acquire(counter: Counter): Id[counter.ReleasableResource] = counter.create
+  override protected def release(counter: Counter)(resource: counter.ReleasableResource): Id[Unit] = resource.close
+  override protected def await[A](a: Id[A]): A = a
+}
+private sealed abstract class ReleasablePoolTest[M[_] : Monad] extends WordSpec {
+  protected def acquire(counter: Counter):M[counter.ReleasableResource]
+  protected def release(counter: Counter)(resource:counter.ReleasableResource):M[Unit]
+  protected def await[A](a:M[A]):A
+  "releasable pool" when {
+    "resource is gotten" should {
+      "create new resource" in {
+        val counter = new Counter
+        val releasablePool: ReleseablePool[M, counter.ReleasableResource, Unit] = createReleseablePool(counter)
+        val resource = await(releasablePool.get(())).right.get
+        resource.isClosed shouldBe false
+        counter.count shouldBe 1
+      }
+      "resource can be closed"in {
+        val counter = new Counter
+        val releasablePool: ReleseablePool[M, counter.ReleasableResource, Unit] = createReleseablePool(counter)
+        val resource = await(releasablePool.get(())).right.get
+        resource.isClosed shouldBe false
+        counter.count shouldBe 1
+        resource.close
+        resource.isClosed shouldBe true
+        counter.count shouldBe 0
+      }
+      "close on pool closes resource"in {
+        val counter = new Counter
+        val releasablePool: ReleseablePool[M, counter.ReleasableResource, Unit] = createReleseablePool(counter)
+        val resource = await(releasablePool.get(())).right.get
+        resource.isClosed shouldBe false
+        counter.count shouldBe 1
+        await(releasablePool.close)
+        resource.isClosed shouldBe true
+        counter.count shouldBe 0
+      }
+      "close on pool closes all resources"in {
+        val counter = new Counter
+        val releasablePool: ReleseablePool[M, counter.ReleasableResource, Unit] = createReleseablePool(counter)
+        val resource1 = await(releasablePool.get(())).right.get
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://app.codacy.com/manual/scarduzio/elasticsearch-readonlyrest-plugin/pullRequest?prid=5113380)